docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,663 @@
+
+### Architecture for Mint/Grant/Burn
+
+![accessControl](/img/mgb-flow.png)
+
+## 2) Lit Action to define swap conditions
+
+For a Swap Lit Action, both parties need to deposit their funds to the PKP and based on the conditions the PKP should be completing the swap:
+
+- Alice (or Bob) creates a Mint/Grant/Burn PKP
+- Alice deposits 4 token A on Base Sepolia to the PKP
+- Bob deposits 8 token B on Chronicle to the PKP
+- Alice (or Bob) executes the Lit Action.
+
+When the Lit Action is executed, each Lit node verifies that the conditions are met before provisioning their key share and executing the swap using the PKP
+
+If the swap conditions aren't met, the Lit Action execution will revert and the tokens will be returned to the initial depositors (Alice and Bob)
+
+Now we'll write the Lit Action that will be used to set the specific swap conditions that will be followed.
+
+Below are the parameters for the cross-chain swap. In the example below, we'll be transferring 4 tokens A on Sepolia for 8 tokens B on Chronicle Yellowstone. These tokens are minted by a custom contract.
+
+```js
+const chainAParams = {
+    from: "0x48e6a467852Fa29710AaaCDB275F85db4Fa420eB",
+    to: "0x291B0E3aA139b2bC9Ebd92168575b5c6bAD5236C",
+    tokenAddress: "0xad50f302a957C165d865eD398fC3ca5A5A2cDA85",
+    chain: "baseSepolia",
+    amount: "4",
+    decimals: 18,
+    chainId: 84532,
+};
+
+const chainBParams = {
+    from: "0x291B0E3aA139b2bC9Ebd92168575b5c6bAD5236C",
+    to: "0x48e6a467852Fa29710AaaCDB275F85db4Fa420eB",
+    tokenAddress: "0x2dcA1a80c89c81C37efa7b401e2d20c1ED99C72F",
+    chain: "yellowstone",
+    amount: "8",
+    decimals: 18,
+    chainId: 175188,
+};
+/* 
+deposit1: wallet X deposits on chain B, if action executes, funds are transferred to wallet Y
+deposit2: wallet Y deposits on chain A, if action executes, funds are transferred to wallet X
+*/
+```
+:::info
+You can use any of the chains [supported](../resources/supported-chains) by Lit when setting your own swap parameters
+:::
+
+The following code can be used to generate the swap Lit Action based on the parameters above: [swapActionGenerator.js](https://github.com/anshss/lit-evm-swap/blob/main/lit/swapActionGenerator.js)
+
+Remember, once a Mint/Grant/Burn PKP is created, anyone can execute the Lit Action on the Lit Network. This execution needs to define the parameters needed by the Lit Action to run. While generating a sign or checking conditions on the action, we need information around the pkp for which we are doing it. Due to this, we need to construct our Lit Action in a way that works with any PKP.
+
+Our Lit Action primarily focuses on three factors:
+
+- Access control conditions for checking funds on both chains
+- Transaction objects for transferring funds between swap parties
+- Clawback transaction objects to revert transferred funds back to their owners if the swap parameters aren't met
+- Lit action conditions which decide which transaction object to sign and return
```

Comment:
"[Lit Action conditions](https://developer.litprotocol.com/sdk/access-control/lit-action-conditions) to determine which transaction objects should be signed and returned"